### ISSUE TITLE
Cluster node stuck in remove state

### ISSUE THREAD

**Issue by josmo** [rancher/rancher#13283]
While accidentally deleting a node that was being registered, the node state is stuck which kills the ability to interact with the cluster.

The UI has 

```This cluster is currently Unavailable; areas that interact directly with it will not be available until the API is ready. Failed to communicate with API server: Get https://x.x.x.x:6443/api/v1/componentstatuses: net/http: request canceled (Client.Timeout exceeded while awaiting headers)```

While the node says it's in the "removing" state with the following information
```Waiting to register with Kubernetes; waiting on user-node-remove_c-v2j5v```

I would expect that you shouldn't be able to get in this state in the first place and is there a work around to still interact with the cluster and remove the node a different way?

---
| Useful | Info |
| :-- | :-- |
|Versions|Rancher `v2.0.0` UI: `v2.0.41` |
|Access|`` `admin`|
|Route|`authenticated.cluster.nodes.index`|

**Comment by pjmazenot** [comment]
Same issue here with **v2.0.2** and a custom cluster.

Cleaning, rebooting re-registering the node didn't fix the problem.

The rest of the cluster is down in the meantime... Anyone found a workaround?

**Comment by smiller393** [comment]
Having the same problem running 2.0.0 on-prem custom cluster. Not finding a way to remove the node so that I can re-add it without removing the whole cluster.

**Comment by hadirsa** [comment]
Have the same problem.

**Comment by phpjob** [comment]
Got the same problem, is there any fix for this or do I have to reinstall the cluster?

**Comment by BrianChristie** [comment]
I had the same issue, it seems to be resolved in v2.0.6

**Comment by felixschul** [comment]
I experienced the same problem in 2.0.6, I do not think this is fixed.

**Comment by frekele** [comment]
I have a similar issue. https://github.com/rancher/rancher/issues/15039

**Comment by loganhz** [comment]
Thanks for the report!
I can not reproduce this issue in v2.1.1
Please let us know in a comment if you can still reproduce the issue, and we'll reopen it

**Comment by loganhz** [comment]
@gxl191911 Can you try to restart node-agent pod?

**Comment by loganhz** [comment]
@gxl191911 Please create a new issue if you still see this issue in 2.1.1

**Comment by gxl191911** [comment]
> @gxl191911 Can you try to restart node-agent pod?

Yes,I Can. But it has caused me a little impact

**Comment by gxl191911** [comment]
> @gxl191911 Please create a new issue if you still see this issue in 2.1.1

I have created a new issue in 2.1.1

**Comment by 3RX0** [comment]
Still an issue in 2.5.12
